src/particle/pdgid/functions.py

```diff
@@ -497,7 +528,11 @@ def three_charge(pdgid):
 
     if _extra_bits(pdgid) > 0:
         if is_nucleus(pdgid):  # ion
-            return 3 * Z(pdgid)
+            Z_pdgid = Z(pdgid)
+            if Z_pdgid is None:
```

Comment:
This in principle should never happen for a nucleus, by construction ;-). But the protection is harmless and avoids trouble if for some reason info was to miss for a given nucleus, so go with it 👍 .

Reply:
You can also state "assert <thing that is always true>" to make MyPy happy. But each function must fill out all logical possibilities, which is nicer to read and verify for someone not looking at everything at once, so I think it's also better. It also does sometimes catch crashes before they happen!

Reply:
Indeed. Makes complete sense.

I have to learn how to use MyPy…!

Reply:
I will probably be writing a post over it so will ping you when I do that. :)

Reply:
I appreciate that, thanks!